android-components/components/service/location/src/test/java/mozilla/components/service/location/MozillaLocationServiceTest.kt

```diff
@@ -274,4 +276,124 @@ class MozillaLocationServiceTest {
             verify(client).fetch(any())
         }
     }
+
+    @Test
+    fun `WHEN fetching region and the cache is valid THEN request is not sent again`() = runTest {
+        val clock = FakeClock()
+
+        run {
+            val client: Client = mock()
+            val response = Response(
+                url = "http://example.org",
+                status = 200,
+                headers = MutableHeaders(),
+                body = Response.Body("{\"country_name\": \"Nepal\", \"country_code\": \"NP\"}".byteInputStream()),
+            )
+            doReturn(response).`when`(client).fetch(any())
+
+            val service = MozillaLocationService(
+                testContext,
+                client,
+                apiKey = "test",
+                currentTime = clock::time,
+            )
+            val region = service.fetchRegion(readFromCache = true)
+
+            assertNotNull(region!!)
+
+            assertEquals("NP", region.countryCode)
+            assertEquals("Nepal", region.countryName)
```

Comment:
How about asserting the region object as it's a data class, instead of 3 individual assertions?

```
val expected = Region(countryCode = "NP", countryName = "Nepal")
assertEquals(expected, region)
```